arch/xtensa/src/esp32/esp32_gpio.c

```diff
@@ -222,12 +222,12 @@ int esp32_configgpio(int pin, gpio_pinattr_t attr)
     }
   else
     {
-      func |= (uint32_t)((2 >> FUNCTION_SHIFT) << MCU_SEL_S);
+      func |= (uint32_t)(PIN_FUNC_GPIO << MCU_SEL_S);
     }
 
   if ((attr & OPEN_DRAIN) != 0)
     {
-      cntrl = (1 << GPIO_PIN_PAD_DRIVER_S);
+      cntrl |= (1 << GPIO_PIN_PAD_DRIVER_S);
```

Comment:
The value here is only used one time and I don't think it will ever be changed from this function in the future.  The rest of the fields are related to the pin's interrupt.  But I understand if it's here for convenience.